From: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Date: Thu, 19 Oct 2006 20:27:39 +0000 (-0700)
Subject: [PATCH] libata: use correct map_db values for ICH8
X-Git-Tag: v2.6.19-rc3~12^2~4
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=158f30c8945fea7cf0d0161cd9463cf2f3d2c19e

[PATCH] libata: use correct map_db values for ICH8

Use valid values for ICH8 map_db.  With the old values, when the
controller was in Native mode, and SCC was 1 (drives configured for
IDE), any drive plugged into a slave port was not recognized.  For
Combined Mode (and SCC is still 1), 2 is a value value for MAP.map_value,
and needs to be recognized.

Signed-off-by:  Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Jeff Garzik <jeff@garzik.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

--- linux-2.6-2.6.18.dfsg.1.orig/drivers/scsi/ata_piix.c	2007-09-12 18:47:41.000000000 -0600
+++ linux-2.6-2.6.18.dfsg.1/drivers/scsi/ata_piix.c	2007-09-12 18:49:09.000000000 -0600
@@ -373,9 +373,9 @@ static const struct piix_map_db ich8_map
 	.present_shift = 8,
 	.map = {
 		/* PM   PS   SM   SS       MAP */
-		{  P0,  NA,  P1,  NA }, /* 00b (hardwired) */
+		{  P0,  P2,  P1,  P3 }, /* 00b (hardwired when in AHCI) */
 		{  RV,  RV,  RV,  RV },
-		{  RV,  RV,  RV,  RV }, /* 10b (never) */
+		{  IDE,  IDE,  NA,  NA }, /* 10b (IDE mode) */
 		{  RV,  RV,  RV,  RV },
 	},
 };
